### PR TITLE
Adding Keybinding Descriptions by Wendy from RPI

### PR DESCRIPTION
**Pull request by hux8**
I added the chart that contains all the implemented keybindings and the description to each action.
-- from RPI Capstone 2019F

## Summary of the Pull Request

Adding the chart that contains all keybindings and description of each action. 
-- From RPI Capstone 2019F (Windows Terminal Keybinding UI team)

 
## PR Checklist
* [ ] Closes #xxx
* [ ] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: doc/cascadia/SettingsSchema.md [hunk 1]
```diff
@@ -89,54 +89,56 @@ Properties listed below are specific to each custom key binding.
...
+| `duplicateTab` | Make a copy of an existing tab. |
+| `newTabProfile0` | Open a new tab with the same profile as tab0. |
+| `newTabProfile1` | Open a new tab with the same profile as tab1. |
+| `newTabProfile2` | Open a new tab with the same profile as tab2. |
+| `newTabProfile3` | Open a new tab with the same profile as tab3. |
+| `newTabProfile4` | Open a new tab with the same profile as tab4. |
+| `newTabProfile5` | Open a new tab with the same profile as tab5. |
+| `newTabProfile6` | Open a new tab with the same profile as tab6. |
+| `newTabProfile7` | Open a new tab with the same profile as tab7. |
+| `newTabProfile8` | Open a new tab with the same profile as tab8. |
```
Comment (carlos-zamora) [inline doc/cascadia/SettingsSchema.md]:
```
    "NewTabAction": {
      "description": "Arguments corresponding to a New Tab Action",
      "allOf": [
        { "$ref": "#/definitions/ShortcutAction" },
        {
          "properties": {
            "action": { "type":"string", "pattern": "newTab" },
            "index": {
              "type": "integer",
              "description": "The index in the new tab dropdown to open in a new tab"
            }
          }
        }
      ]
    },
```

File: doc/cascadia/SettingsSchema.md [hunk 2]
```diff
@@ -89,54 +89,56 @@ Properties listed below are specific to each custom key binding.
...
-- resizePaneDown
-- moveFocusLeft
-- moveFocusRight
-- moveFocusUp
-- moveFocusDown
-- toggleFullscreen
+| Action | Description |
+| -------- | ----------- |
+| `copy` | Make a duplicate of the selected content. |
+| `copyTextWithoutNewlines` | Make a duplicate of the selected content and discard the newline characters. |
```
Comment (carlos-zamora) [inline doc/cascadia/SettingsSchema.md]:
```
    "CopyAction": {
      "description": "Arguments corresponding to a Copy Text Action",
      "allOf": [
        { "$ref": "#/definitions/ShortcutAction" },
        {
          "properties": {
            "action": { "type": "string", "pattern": "copy" },
            "copyNewlines": {
              "type": "boolean",
              "default": true,
              "description": "If false, concatenates the copied text to one line."
            }
          }
        }
      ]
    },
```
This one is _very temporarily_ blocked by #3811 but that should be going in soon so I wouldn't worry about it.

File: doc/cascadia/SettingsSchema.md [hunk 3]
```diff
@@ -89,54 +89,56 @@ Properties listed below are specific to each custom key binding.
...
+| `scrollDownPage` | Move the screen down a whole page to the later content of the tab. |
+| `switchToTab0` | Change the content of the current window to tab 0. |
+| `switchToTab1` | Change the content of the current window to tab 1. |
+| `switchToTab2` | Change the content of the current window to tab 2. |
+| `switchToTab3` | Change the content of the current window to tab 3. |
+| `switchToTab4` | Change the content of the current window to tab 4. |
+| `switchToTab5` | Change the content of the current window to tab 5. |
+| `switchToTab6` | Change the content of the current window to tab 6. |
+| `switchToTab7` | Change the content of the current window to tab 7. |
+| `switchToTab8` | Change the content of the current window to tab 8. |
```
Comment (carlos-zamora) [inline doc/cascadia/SettingsSchema.md]:
```
    "SwitchToTabAction": {
      "description": "Arguments corresponding to a Switch To Tab Action",
      "allOf": [
        { "$ref": "#/definitions/ShortcutAction" },
        {
          "properties": {
            "action": { "type": "string", "pattern": "switchToTab" },
            "index": {
              "type": "integer",
              "default": 0,
              "description": "Which tab to switch to, with the first being 0"
            }
          }
        }
      ],
      "required": [ "index" ]
    },

File: doc/cascadia/SettingsSchema.md [hunk 4]
```diff
@@ -89,54 +89,56 @@ Properties listed below are specific to each custom key binding.
...
+| `openSettings` | Open the settings page. |
+| `splitPane` | Separate the pane into two. |
+| `resizePaneLeft` | Change the size of the left pane. |
+| `resizePaneRight` | Change the size of the right pane. |
+| `resizePaneUp` | Change the size of the top pane. |
+| `resizePaneDown` | Change the size of the bottom pane. |
+| `moveFocusLeft` | Changes the focus direction to left. |
+| `moveFocusRight` | Changes the focus direction to right. |
+| `moveFocusUp` | Changes the focus direction to up. |
+| `moveFocusDown` | Changes the focus direction to down. |
```
Comment (carlos-zamora) [inline doc/cascadia/SettingsSchema.md]:
```
    "MoveFocusAction": {
      "description": "Arguments corresponding to a Move Focus Action",
      "allOf": [
        { "$ref": "#/definitions/ShortcutAction" },
        {
          "properties": {
            "action": { "type": "string", "pattern": "moveFocus" },
            "direction": {
              "$ref": "#/definitions/Direction",
              "default": "left",
              "description": "The direction to move focus in, between panes"
            }
          }
        }
      ],
      "required": [ "direction" ]
    },

File: doc/cascadia/SettingsSchema.md [hunk 5]
```diff
@@ -89,54 +89,56 @@ Properties listed below are specific to each custom key binding.
...
+| `switchToTab5` | Change the content of the current window to tab 5. |
+| `switchToTab6` | Change the content of the current window to tab 6. |
+| `switchToTab7` | Change the content of the current window to tab 7. |
+| `switchToTab8` | Change the content of the current window to tab 8. |
+| `openSettings` | Open the settings page. |
+| `splitPane` | Separate the pane into two. |
+| `resizePaneLeft` | Change the size of the left pane. |
+| `resizePaneRight` | Change the size of the right pane. |
+| `resizePaneUp` | Change the size of the top pane. |
+| `resizePaneDown` | Change the size of the bottom pane. |
```
Comment (carlos-zamora) [inline doc/cascadia/SettingsSchema.md]:
```
    "ResizePaneAction": {
      "description": "Arguments corresponding to a Resize Pane Action",
      "allOf": [
        { "$ref": "#/definitions/ShortcutAction" },
        {
          "properties": {
            "action": { "type": "string", "pattern": "resizePane" },
            "direction": {
              "$ref": "#/definitions/Direction",
              "default": "left",
              "description": "The direction to move the pane separator in"
            }
          }
        }
      ],
      "required": [ "direction" ]
    },

File: doc/cascadia/SettingsSchema.md [hunk 6]
```diff
@@ -89,54 +89,56 @@ Properties listed below are specific to each custom key binding.
...
+| `switchToTab1` | Change the content of the current window to tab 1. |
+| `switchToTab2` | Change the content of the current window to tab 2. |
+| `switchToTab3` | Change the content of the current window to tab 3. |
+| `switchToTab4` | Change the content of the current window to tab 4. |
+| `switchToTab5` | Change the content of the current window to tab 5. |
+| `switchToTab6` | Change the content of the current window to tab 6. |
+| `switchToTab7` | Change the content of the current window to tab 7. |
+| `switchToTab8` | Change the content of the current window to tab 8. |
+| `openSettings` | Open the settings page. |
+| `splitPane` | Separate the pane into two. |
```
Comment (carlos-zamora) [inline doc/cascadia/SettingsSchema.md]:
```
    "SplitPaneAction": {
      "description": "Arguments corresponding to a Split Pane Action",
      "allOf": [
        { "$ref": "#/definitions/ShortcutAction" },
        {
          "properties": {
            "action": { "type": "string", "pattern": "splitPane" },
            "split": {
              "$ref": "#/definitions/SplitState",
              "default": "vertical",
              "description": "The orientation to split the pane in, either vertical (think [|]) or horizontal (think [-])"
            }
          }
        }
      ],
      "required": [ "split" ]
    },

**Comment by nahiyan8** [comment]
Could you please add a short description for the key chord definitions as well? It's not too clear what a user is supposed to write for all the special keys (e.g. there's exactly `numpad_5` or there's `esc` and not `escape`, etc.) or whether we are supposed to separate parts with `+` or `-`. Also, the JSON schema helps a bit but this information should really be documented!

The parsing code can be found in: `src/cascadia/TerminalApp/AppKeyBindingsSerialization.cpp`

Here's my take on the doc for all the special keys, with descriptions sourced from [the dev center](https://docs.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes):

> The string should fit the format `"[ctrl+][alt+][shift+]<keyName>"`, where each modifier is optional,  separated by `+` symbols, and `keyName` is either one of the names listed in the table below, or is one of `0-9a-zA-Z`. The string should be written in full lowercase.

| `keyName` String | Description |
|--------------------|--------------|
| `backspace` | BACKSPACE key |
| `tab` | TAB key |
| `enter` | ENTER key |
| `esc` | ESC key |
| `space` | SPACEBAR |
| `pgup` | PAGE UP key |
| `pgdn` | PAGE DOWN key |
| `end` | END key |
| `home` | HOME key |
| `left` | LEFT ARROW key |
| `up` | UP ARROW key |
| `right` | RIGHT ARROW key |
| `down` | DOWN ARROW key |
| `insert` | INS key |
| `delete` | DEL key |
| `numpad_0` to `numpad_9` | Numeric keypad keys (0 to 9) |
| `numpad_multiply` | Multiply key |
| `numpad_plus` | Add key |
| `numpad_minus` | Subtract key |
| `numpad_period` | Decimal key |
| `numpad_divide` | Divide key |
| `f1` to `f24` | F1-F24 keys |
| `plus` | For any country/region, the '+' key |

Note: The last `plus` key description seemed weird. From my testing, it's the `=`/`+` key as usual. But the code name for it is `VK_OEM_PLUS` and shares the same description as other `OEM` keys. They are apparently called `OEM` because they provide compatibility between different keyboard layouts. In this case, it's probably here to prevent issues with keybindings such as `ctrl++`.

**Comment by carlos-zamora** [comment]
> Could you please add a short description for the key chord definitions as well? It's not too clear what a user is supposed to write for all the special keys (e.g. there's `numpad_5`, there's `esc` and not `escape`, etc.) or whether we are supposed to separate parts with `+` or `-`. Also, the JSON schema helps a bit but this information should really be documented!
> 
> The parsing code can be found in: `src/cascadia/TerminalApp/AppKeyBindingsSerialization.cpp`
> 
> Here's my take on the doc for all the special keys, adapted from the code with descriptions sourced from [the dev center](https://docs.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes):
> 
> > The string should fit the format `"[ctrl+][alt+][shift+]<keyName>"`, where each modifier is optional,  separated by `+` symbols, and `keyName` is either one of the names listed in the table below, or is one of `0-9a-zA-Z`. The string should be written in full lowercase.
> 
> `keyName` String	Description
> `backspace`	BACKSPACE key
> `tab`	TAB key
> `enter`	ENTER key
> `esc`	ESC key
> `space`	SPACEBAR
> `pgup`	PAGE UP key
> `pgdn`	PAGE DOWN key
> `end`	END key
> `home`	HOME key
> `left`	LEFT ARROW key
> `up`	UP ARROW key
> `right`	RIGHT ARROW key
> `down`	DOWN ARROW key
> `insert`	INS key
> `delete`	DEL key
> `numpad_0` to `numpad_9`	Numeric keypad keys (0 to 9)
> `numpad_multiply`	Multiply key
> `numpad_plus`	Add key
> `numpad_minus`	Subtract key
> `numpad_period`	Decimal key
> `numpad_divide`	Divide key
> `f1` to `f24`	F1-F24 keys
> `plus`	For any country/region, the '+' key
> Note: The last `plus` key description seemed weird. From my testing, it's the `=`/`+` key as usual. But the code name for it is `VK_OEM_PLUS` and shares the same description as other `OEM` keys. They are apparently called `OEM` because they provide compatibility between different keyboard layouts. In this case, it's probably here to prevent issues with keybindings such as `ctrl++`.

Created a new issue for this since it's a little out of scope. Continuing discussion on #3856.

**Comment by zadjii-msft** [comment]
(this should probably not be closed)

**Comment by carlos-zamora** [comment]
Closing since most of the feedback is incorporated in #4392. Thanks @hux8!

**Pull request closed without merging**